test/src/helpers/packTester.ts

```diff
@@ -120,7 +121,9 @@ export async function assertPack(fixtureName: string, packagerOptions: PackagerO
 
       if (checkOptions.isInstallDepsBefore) {
         // bin links required (e.g. for node-pre-gyp - if package refers to it in the install script)
-        await spawn(process.platform === "win32" ? "npm.cmd" : "npm", ["install", "--production", "--legacy-peer-deps"], {
+        const pm = await detect({ cwd: projectDir })
+        let cmd = process.platform === "win32" ?  pm+".cmd":pm
```

Comment:
Please run `pnpm prettier` when ready for review 🙂

Reply:
No problem, I've been using `pnpm lint --fix`, but it seems this didn't format this code.

Reply:
On windows, there's an error when running `pnpm prettier`.
![image](https://github.com/user-attachments/assets/fe21a3b8-3f84-44a7-8a04-a11348a5c336)